src/controller/java/AndroidCallbacks.cpp

```diff
@@ -576,18 +617,19 @@ CHIP_ERROR InvokeCallback::CreateInvokeElement(const app::ConcreteCommandPath &
         err = TlvToJson(readerForJson, json);
         ReturnErrorOnFailure(err);
         UtfString jsonString(env, json.c_str());
-        outObj = env->CallStaticObjectMethod(invokeElementCls, invokeElementCtor, static_cast<jint>(aPath.mEndpointId),
-                                             static_cast<jlong>(aPath.mClusterId), static_cast<jlong>(aPath.mCommandId),
-                                             jniByteArray.jniValue(), jsonString.jniValue());
+        localRef = env->CallStaticObjectMethod(invokeElementCls, invokeElementCtor, static_cast<jint>(aPath.mEndpointId),
+                                               static_cast<jlong>(aPath.mClusterId), static_cast<jlong>(aPath.mCommandId),
+                                               jniByteArray.jniValue(), jsonString.jniValue());
     }
     else
     {
-        outObj = env->CallStaticObjectMethod(invokeElementCls, invokeElementCtor, static_cast<jint>(aPath.mEndpointId),
-                                             static_cast<jlong>(aPath.mClusterId), static_cast<jlong>(aPath.mCommandId), nullptr,
-                                             nullptr);
+        localRef = env->CallStaticObjectMethod(invokeElementCls, invokeElementCtor, static_cast<jint>(aPath.mEndpointId),
+                                               static_cast<jlong>(aPath.mClusterId), static_cast<jlong>(aPath.mCommandId), nullptr,
+                                               nullptr);
     }
+    VerifyOrReturnError(localRef != nullptr, CHIP_JNI_ERROR_NULL_OBJECT);
+    outObj = env->NewGlobalRef(localRef);
```

Comment:
This seems really suspect; that global ref does not seem to be cleaned up in the caller.  If you're not creating a new JNI frame here, it would be vastly cleaner to not allocate a new global ref; you're running in someone else's JNI frame

Reply:
https://github.com/project-chip/connectedhomeip/pull/29710